proctor-common/src/test/java/com/indeed/proctor/common/TestTestDependencies.java

```diff
@@ -167,6 +168,59 @@ public void testValidateDependencyAndReturnErrorReason_negativeBucket() {
         )).hasValue("A test example_tst depends on negative bucket value -1 of parent_tst");
     }
 
+    @Test
+    public void testValidateDependencyAndReturnErrorReason_validWithDifferentTestType() {
+        final ConsumableTestDefinition definition = ConsumableTestDefinition.fromTestDefinition(
+                stubTestDefinition(TEST_NAME)
```

Comment:
nits: You might want to call `setTestType` with some value like `TestType.ANONYMOUS_USER` explicitly to make clear what test type is used (rather than relying on the default).